apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerClient.java

```diff
@@ -245,4 +256,10 @@ int getErrorCount() {
         T withConnection(HttpURLConnection connection) throws IOException;
     }
 
+    public synchronized void overrideApmServerUrls(List<URL> serverUrls) {
+        logger.debug("server_urls override with value = ({}).", serverUrls);
+        this.serverUrls = serverUrls;
```

Comment:
Shuffle the URLs by reusing the `shuffleUrls` method. To achieve that, change the parameter of `shuffleUrls` from `ReporterConfiguration` to `List<URL>`.

Reply:
added